### PR TITLE
Change all URIFromDestinationV1 to AddressableFromDestinationV1

### LINKED ISSUE #3109: Eventing TLS: migrate all resolver.URIResolver usages over to AddressableFromDestinationV1
**Problem**

This is the same issue as https://github.com/knative/eventing/issues/6878 but for components in this repository

**Exit Criteria**

migrate all resolver.URIResolver usages over to AddressableFromDestinationV1.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1

**Additional context (optional)**

- similar PR: https://github.com/knative/eventing/pull/6883

/good-first-issue

### PR DESCRIPTION
**Pull request by Leo6Leo**
Fixes #3109 

## Proposed Changes

- URIFromDestinationV1 and AddressableFromDestinationV1 are 2 function do the same thing, but their return types are different. URIFromDestinationV1 will only URL string, however, AddressableFromDestinationV1 will return a `Destination` object, which contains more information (e.g CAcert)

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/consumer/consumer.go [hunk 1]
```diff
@@ -269,12 +269,12 @@ func (r *Reconciler) reconcileReplyStrategy(ctx context.Context, c *kafkainterna
...
 	}
 	if c.Spec.Reply.URLReply != nil && c.Spec.Reply.URLReply.Enabled {
-		destination, err := r.Resolver.URIFromDestinationV1(ctx, c.Spec.Reply.URLReply.Destination, c)
+		destination, err := r.Resolver.AddressableFromDestinationV1(ctx, c.Spec.Reply.URLReply.Destination, c)
 		if err != nil {
 			return fmt.Errorf("failed to resolve reply destination: %w", err)
 		}
 		egress.ReplyStrategy = &contract.Egress_ReplyUrl{
-			ReplyUrl: destination.String(),
+			ReplyUrl: destination.URL.String(),
```
Comment (Vishal-Chdhry) [inline control-plane/pkg/reconciler/consumer/consumer.go:277]: You have to make this change in a few other places as well

https://github.com/knative-sandbox/eventing-kafka-broker/pull/3114/files#diff-6ee4865953a537188757b144b2c85c1d9fb9cf0b206c660216e53289f2e98f2fR155
https://github.com/knative-sandbox/eventing-kafka-broker/pull/3114/files#diff-15d344ea3208a104992f6e8ee1431a4825c26171d1038fac81fddeaa5390bef1R320

**Review by Vishal-Chdhry — CHANGES_REQUESTED** [review]

File: control-plane/pkg/core/config/utils.go [hunk 2]
```diff
@@ -66,11 +66,11 @@ func EgressConfigFromDelivery(
 		if destination.Ref != nil && destination.Ref.Namespace == "" {
 			destination.Ref.Namespace = parent.GetNamespace()
 		}
-		deadLetterSinkURL, err := resolver.URIFromDestinationV1(ctx, *delivery.DeadLetterSink, parent)
+		deadLetterSinkURL, err := resolver.AddressableFromDestinationV1(ctx, *delivery.DeadLetterSink, parent)
```
Comment (Vishal-Chdhry) [inline control-plane/pkg/core/config/utils.go:69]:
```suggestion
		deadLetterSinkAddr, err := resolver.AddressableFromDestinationV1(ctx, *delivery.DeadLetterSink, parent)
```

NIT, update variable names, as it is a `addressable` now and not a `URL`

File: control-plane/pkg/reconciler/consumer/consumer.go [hunk 3]
```diff
@@ -133,11 +133,11 @@ func (r *Reconciler) reconcileContractResource(ctx context.Context, c *kafkainte
 }
 
 func (r *Reconciler) reconcileContractEgress(ctx context.Context, c *kafkainternals.Consumer) (*contract.Egress, error) {
-	destination, err := r.Resolver.URIFromDestinationV1(ctx, c.Spec.Subscriber, c)
+	destination, err := r.Resolver.AddressableFromDestinationV1(ctx, c.Spec.Subscriber, c)
```
Comment (Vishal-Chdhry) [inline control-plane/pkg/reconciler/consumer/consumer.go:136]:
```suggestion
	destinationAddr, err := r.Resolver.AddressableFromDestinationV1(ctx, c.Spec.Subscriber, c)
```

the same change in all other places in `consumer/consumer.go` and `trigger/trigger.go`

**Review by Vishal-Chdhry — CHANGES_REQUESTED** [review]

**Comment by Vishal-Chdhry** [comment]
maybe @pierDipi knows the answer, afaik, tests are run on every pr everytime a change is made, but that should not be a problem. even hold won't stop testing (lgtm will remove the hold)

**Comment by pierDipi** [comment]
I don't think we can stop it :)

it will run the tests when you push new changes

**Comment by pierDipi** [comment]
Yes

/approve

**Comment by matzew** [comment]
> I don't think we can stop it :)

> it will run the tests when you push new changes

Perhaps "DRAFT" PRs do not have that? But I am not sure, and would not too much worry about that

**Comment by creydr** [comment]
> I pulled up a draft PR [here](https://github.com/knative-sandbox/eventing-kafka-broker/pull/3123). It seems like some tests are suspended, some tests are still running. @matzew

Usually the Github actions run always. Only the Prow jobs will not run on draft PRs.

**Comment by pierDipi** [comment]
I'm also not sure if you start with ready PR and then convert to draft (which was the case in this PR), does it stop?

**Pull request merged**